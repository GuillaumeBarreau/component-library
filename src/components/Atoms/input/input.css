/* Input */
.input {
   background-color: #f0eeee;
   font-size: 1.4rem;
   border-radius: 1.25rem;
   margin: 0.5rem;
   padding: 1rem;
   border-radius: 4px;
}