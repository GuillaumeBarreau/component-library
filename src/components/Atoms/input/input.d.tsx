export interface ComponentInputProps {
  className?: string;
  handleOnChange: React.ChangeEventHandler<HTMLInputElement>;
  type?: "text"
  placeholder?: string;
  value: string | undefined;
}