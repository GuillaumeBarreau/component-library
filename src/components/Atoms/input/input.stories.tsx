import React from "react";
import {Input} from "./input";
import { Meta } from "@storybook/react/types-6-0";
import { ComponentInputProps } from "./input.d";
import "../../_settings/colors/colors.css";

export default {
  component: Input,
  title: "Input",
  argTypes: {
    className: {
      description: "className content",
      control: "text",
    },
  },
  parameters: {
    a11y: {
      // optional selector which element to inspect
      element: "#root",
      // optional flag to prevent the automatic check
      manual: false,
    },
  },
} as Meta;

const Template: any = (
  args: JSX.IntrinsicAttributes & ComponentInputProps
) => <>
  <section style={{display: "flex", justifyContent: "center"}}>
    <Input {...args} />
  </section>
</>;

export const InputDefault = Template.bind({});
InputDefault.args = {
  placeholder: "input",
  handleOnChange: () => null,
};
