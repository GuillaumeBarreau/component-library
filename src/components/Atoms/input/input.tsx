import React from "react";
import classNames from "classnames";
import { ComponentInputProps } from "./input.d";
import "./input.css";

export const Input: React.FC<ComponentInputProps> = (props) => {
  const {
    className,
    handleOnChange,
    placeholder,
    type = "text",
    namedInput,
    labelDescription,
    value,
  } = props;
  const prefixCls: string = "input";

  const classes = classNames(
    prefixCls,
    {
      [`${prefixCls}-${type}`]: type,
    },
    className
  );

  const inputNode = (
    <>
      <label htmlFor={namedInput}>{labelDescription}</label>
      <input
        id={namedInput}
        className={classes}
        onChange={handleOnChange}
        type={type}
        placeholder={placeholder}
        value={value}
      />
    </>
  );

  return inputNode;
};
