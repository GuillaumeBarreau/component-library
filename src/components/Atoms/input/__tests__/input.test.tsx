import React from "react";
import { shallow } from "enzyme";
import { Input } from "../input";
import { ComponentInputProps } from "../input.d";

describe("Input", () => {
  let defaultProps: JSX.IntrinsicAttributes & ComponentInputProps;

  beforeEach(() => {
    defaultProps = {
      className: "",
      labelDescription: "label description",
      namedInput: "namedInput",
      handleOnChange: jest.fn(),
      type: "text",
      placeholder: "Login",
      value: "value",
    };
  });

  it("renders correctly", () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("mount correctly", () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
  });

  it("Should contain defaultProps:placeholder correctly", () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain("Login");
  });

  it("Should contain defaultProps:labelDescription correctly", () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain("label description");
  });

  it("Should contain defaultProps:namedInput correctly", () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain("label description");
  });
});
