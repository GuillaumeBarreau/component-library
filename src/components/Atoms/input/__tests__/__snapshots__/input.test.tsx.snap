// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
"<Fragment>
  <label htmlFor=\\"namedInput\\">
    label description
  </label>
  <input id=\\"namedInput\\" className=\\"input input-text\\" onChange={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} type=\\"text\\" placeholder=\\"Login\\" value=\\"value\\" />
</Fragment>"
`;
